File: tests/lua-tests/src/CurrentLanguageTest/CurrentLanguageTest.lua
local function CurrentLanguageTest()
    local ret = cc.Layer:create()
    local  label = cc.LabelTTF:create("Current language Test", "Arial", 28)
    ret:addChild(label, 0)
    label:setPosition( cc.p(VisibleRect:center().x, VisibleRect:top().y-50) )

    local labelLanguage = cc.LabelTTF:create("", "Arial", 20)
    labelLanguage:setPosition(VisibleRect:center())

    local currentLanguageType = cc.Application:getInstance():getCurrentLanguage()

    if currentLanguageType == cc.LANGUAGE_ENGLISH then
        labelLanguage:setString("current language is English")
    elseif currentLanguageType == cc.LANGUAGE_CHINESE then
        labelLanguage:setString("current language is Chinese")
    elseif currentLanguageType == cc.LANGUAGE_FRENCH then
        labelLanguage:setString("current language is French")
    elseif currentLanguageType == cc.LANGUAGE_GERMAN then
        labelLanguage:setString("current language is German")
    elseif currentLanguageType == cc.LANGUAGE_ITALIAN then
        labelLanguage:setString("current language is Italian")
    elseif currentLanguageType == cc.LANGUAGE_RUSSIAN then
        labelLanguage:setString("current language is Russian")
    elseif currentLanguageType == cc.LANGUAGE_SPANISH then
        labelLanguage:setString("current language is Spanish")
    elseif currentLanguageType == cc.LANGUAGE_KOREAN then
        labelLanguage:setString("current language is Korean")
    elseif currentLanguageType == cc.LANGUAGE_JAPANESE then
        labelLanguage:setString("current language is Japanese")
    elseif currentLanguageType == cc.LANGUAGE_HUNGARIAN then
        labelLanguage:setString("current language is Hungarian")
    elseif currentLanguageType == cc.LANGUAGE_PORTUGUESE then
        labelLanguage:setString("current language is Portuguese")
    elseif currentLanguageType == cc.LANGUAGE_ARABIC then
        labelLanguage:setString("current language is Arabic")
    end
    ret:addChild(labelLanguage)
    return ret
end

function CurrentLanguageTestMain()
    local scene = cc.Scene:create()
    local pLayer = CurrentLanguageTest()
    scene:addChild(pLayer)
    scene:addChild(CreateBackMenuItem())
    return scene
end

File: tests/js-tests/src/CurrentLanguageTest/CurrentLanguageTest.js
/****************************************************************************
 Copyright (c) 2008-2010 Ricardo Quesada
 Copyright (c) 2011-2012 cocos2d-x.org
 Copyright (c) 2013-2016 Chukong Technologies Inc.
 Copyright (c) 2017-2018 Xiamen Yaji Software Co., Ltd.

 http://www.cocos2d-x.org

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in
 all copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
 ****************************************************************************/

var CurrentLanguageTest = cc.Layer.extend({
    ctor:function () {
        this._super();

        var s = cc.director.getWinSize();
        var label = new cc.LabelTTF("Current language Test", "Arial", 28);
        this.addChild(label, 0);
        label.x = s.width / 2;
	    label.y = s.height - 50;

        var labelLanguage = new cc.LabelTTF("", "Arial", 20);
        labelLanguage.x = s.width / 2;
	    labelLanguage.y = s.height / 2;

        var currentLanguageType = cc.sys.language;
        switch (currentLanguageType) {
            case cc.sys.LANGUAGE_ENGLISH:
                labelLanguage.setString("current language is English");
                break;
            case cc.sys.LANGUAGE_CHINESE:
                labelLanguage.setString("current language is Chinese");
                break;
            case cc.sys.LANGUAGE_FRENCH:
                labelLanguage.setString("current language is French");
                break;
            case cc.sys.LANGUAGE_GERMAN:
                labelLanguage.setString("current language is German");
                break;
            case cc.sys.LANGUAGE_ITALIAN:
                labelLanguage.setString("current language is Italian");
                break;
            case cc.sys.LANGUAGE_RUSSIAN:
                labelLanguage.setString("current language is Russian");
                break;
            case cc.sys.LANGUAGE_SPANISH:
                labelLanguage.setString("current language is Spanish");
                break;
        }

        this.addChild(labelLanguage);
    }
});

var CurrentLanguageTestScene = TestScene.extend({
    runThisTest:function () {
        if(window.sideIndexBar){
            window.sideIndexBar.changeTest(0, 8);
        }

        var layer = new CurrentLanguageTest();
        this.addChild(layer);

        cc.director.runScene(this);
    }
});

var arrayOfCurrentLanguageTest = [
    CurrentLanguageTest
];